src/node-jspm-jasmine.js

```diff
@@ -141,17 +180,73 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// the "onComplete" hook to prevent jasmine's self-righteous exit
+		jasmine.onComplete(function (passed) {
+			// avoid misalignment with jasmine's output
+			console.log('');
+			// import the rest of the modules not already imported (evaluated)
+			// as dependencies of specs
+			Promise.all(Object.keys(coverage.coverageFiles).map(function (file) {
+				return SystemJS.import(path.join(process.cwd(), file));
+			})).then(function () {
+				if (typeof __coverage__ === 'undefined') {
+					console.log("No coverage was collected for files matching globs " + coverage.files);
+				} else {
+					try {
+						const collector = remap(__coverage__)
+						let report = Report.create(
+							(coverage.reporter || 'html'),
+							{
+								dir: coverage.dir
+							}
+						);
+						report.writeReport(collector, true);
```

Comment:
@joeldenning Try it now and lmk. As far as I know, that issue happens when the source for the file that Istanbul instruments is different from the source that it tries to generate the html report from. Which means one thing; one of your files is getting transpiled into something different from its source, but we with no sourcemap! Run this updated pr and you should see a warning that looks like an error:
`[Error: Could not find source map for: "/path/to/file/that/got/transpiled/without/sourcemap"]`
If this is the case, please comment out the tmp folder deletion code, compare your source file with the one on the tmp folder, and verify that they are different, yet, don't contain a sourcemap.

Assuming all is well, I'm going to do a pr on [remap-istanbul](https://github.com/SitePen/remap-istanbul) to have this "Could not find source map for" warning be optional. [Not sure why it isn't](https://github.com/SitePen/remap-istanbul/commit/c0c97881e8b59f333b4646dd2b0da8944c64decf).

Is this issue present in the master branch also or just this pr?